src/vue/useBaseQuery.ts

```diff
@@ -4,30 +4,24 @@ import {
   readonly,
   ToRefs,
   reactive,
-  watchEffect,
   unref,
+  isRef,
+  watch,
+  UnwrapRef,
 } from "vue-demi";
-
 import type {
   QueryObserver,
   QueryKey,
   QueryObserverOptions,
   QueryObserverResult,
 } from "react-query/core";
-
+import type { QueryFunction } from "react-query/types/core";
+import clonedeepwith from "lodash.clonedeepwith";
```

Comment:
I'm a little worried about usage of this module.
[Bundlephobia](https://bundlephobia.com/package/lodash.clonedeepwith@4.5.0) reports size of 3.3kb for it. It's 1/3 of this library.
Also if you take a look at it's [source code](https://github.com/lodash/lodash/blob/4.5.0-npm-packages/lodash.clonedeepwith/index.js) it contains a lot of unnecessary safeguards and compatibility features.
I mean, we won't copy whole window object under IE :D

I think we can find a smaller utility for this, or even write it.

Reply:
Yeah, I miscalculated the size of it (looked at wrong numbers) :(

Lodash's `clonedeep` is bulletproof solution. 
Many other solutions lack support for Maps/Sets, etc.

I found only one popular proper solution and its npm downloads even exceed the lodash's ones (https://www.npmtrends.com/lodash.clonedeep-vs-clone):

https://github.com/pvorb/clone

It's 1kB (minified and gzipped) https://bundlephobia.com/package/clone@2.1.2

It doesn't provide ability to customize the function, but I could copy the implementation and customize the code to our needs.

WDYT, @DamianOsipiuk ?

Reply:
Hmm, this one looks better.

I still think that we do not need such sophisticated solution as we would copy mostly primitives, functions, arrays and objects. 
Besides `queryKey` options  have quite narrow set of possible values. As for `queryKey`, I'm not sure if people would use Set's or Map's for it (at least i do not see the point). 

Personally i would prefer to cover very basic set of values. And if people start coming with some weird use-cases we could use full-fledged solution. Unless you see a use case for some more advanced values right now.

Reply:
>we would copy mostly primitives, functions, arrays and objects.
>Personally i would prefer to cover very basic set of values.

That's a good point! 👍  
Completely agree! I was overcomplicating the problem. 
Will do the changes